provisioning/README.md

```diff
@@ -1,6 +1,38 @@
 # enigmail-provisioning
 Contains provisioning scripts for developing Enigmail, for Vagrant and Docker
 
-
 If you have problems with permissions on the shared directory, this incantation is useful:
  chcon -Rt svirt_sandbox_file_t $DIR
+
+## Should I setup with Docker or Vagrant?
+[Docker](https://www.docker.com) is a linux only virtual environment tool, which means it is easier to setup on a linux host. Mac OS X and Windows hosts will require the setup of a linux virtual machine for docker to work because it is based on [LinuX Containers (LxC)](https://linuxcontainers.org).
+
+Docker’s virtual environments are light weight versions of virtual machines. They aren’t distinct virtual computers with their own processors and hardware emulation. As a result, virtual environments place only a little extra load on the system. On the other hand, Docker’s isolation is not as good as a full virtual machine’s. Docker allows you to quickly create a specific, semi-isolated environment for short lived testing and debugging.
+
+[Vagrant](https://www.vagrantup.com) is a wrapper on top of a full virtual machine. Vagrant is better when you require better isolation, and it is more convenient to use when you already have vagrant tools and you don’t have docker on your non-linux host machine.
+
+## Quick Setup With Docker
+Please follow your [OS specific instructions](https://docs.docker.com/installation/#installation) to install docker on your system.
+
+Once docker is installed...
+
+1. Start the docker daemon. Those platform specific instructions should be in the docs above, after the installation steps.
+2. `cd provisioning; sudo ./docker-init.sh` - docker-init.sh should be run inside the provisioning folder
```

Comment:
Ah okay. That's why you said to look through the docker installation instructions for Ubuntu specifically because it details how to avoid using sudo.